lan867x/examples/lan867x_tcp_sockets/server/main/server.c

```diff
@@ -0,0 +1,103 @@
+#include "esp_log.h"
+#include "ethernet_init.h"
+#include "lwip/sockets.h"
+
+#define SOCKET_PORT         5000
+#define LISTENER_MAX_QUEUE  10
+#define SOCKET_MAX_LENGTH   128
+#define IP_TO_UINT32(o1,o2,o3,o4)   (o4<<24)+(o3<<16)+(o2<<8)+(o1)
+
+static const char *TAG = "lan867x_example";
+
+/** Event handler for IP_EVENT_ETH_GOT_IP */
+static void got_ip_event_handler(void *arg, esp_event_base_t event_base, int32_t event_id, void *data)
+{
+    ip_event_got_ip_t *event = (ip_event_got_ip_t *) data;
+    const esp_netif_ip_info_t *ip_info = &event->ip_info;
+
+    ESP_LOGI(TAG, "Ethernet Got IP Address");
+    ESP_LOGI(TAG, "~~~~~~~~~~~");
+    ESP_LOGI(TAG, "ETHIP:" IPSTR, IP2STR(&ip_info->ip));
+    ESP_LOGI(TAG, "ETHMASK:" IPSTR, IP2STR(&ip_info->netmask));
+    ESP_LOGI(TAG, "ETHGW:" IPSTR, IP2STR(&ip_info->gw));
+    ESP_LOGI(TAG, "~~~~~~~~~~~");
+}
+
+void my_event_connected_handler(void *esp_netif, esp_event_base_t base, int32_t event_id, void *data)
+{
+    esp_netif_dhcps_start(esp_netif);
+}
+
+void my_ip_assigned_handler(void *esp_netif, esp_event_base_t base, int32_t event_id, void *data)
+{
+    ip_event_got_ip_t *event = (ip_event_got_ip_t *) data;
+    const esp_netif_ip_info_t *ip_info = &event->ip_info;
+    ESP_LOGI(TAG, "Assigned ip " IPSTR " to a connected device.", IP2STR(&ip_info->ip));
+}
+
+void app_main(void)
+{
+    // Create default event loop that running in background
+    ESP_ERROR_CHECK(esp_event_loop_create_default());
+    // Initialize Ethernet driver
+    uint8_t eth_port_cnt = 0;
+    esp_eth_handle_t *eth_handles;
+    ESP_ERROR_CHECK(ethernet_init_all(&eth_handles, &eth_port_cnt));
+    // Initialize TCP/IP network interface aka the esp-netif (should be called only once in application)
+    ESP_ERROR_CHECK(esp_netif_init());
+    esp_netif_ip_info_t ip_info = {
+        .ip = {.addr = IP_TO_UINT32(192, 168, 1, 1)},
+        .netmask = {.addr =  IP_TO_UINT32(255, 255, 255, 0)},
+        .gw = {.addr = IP_TO_UINT32(192, 168, 1, 255)}
+    };
+    const esp_netif_inherent_config_t eth_behav_cfg = {
+            .get_ip_event = IP_EVENT_ETH_GOT_IP,
+            .lost_ip_event = 0,
+            .flags = ESP_NETIF_DHCP_SERVER,
+            .ip_info = &ip_info,
+            .if_key = "ETH_DHCPS",
+            .if_desc = "eth",
+            .route_prio = 50
+    };
+    esp_netif_config_t eth_as_dhcps_cfg = { .base = &eth_behav_cfg, .stack = ESP_NETIF_NETSTACK_DEFAULT_ETH };
+    esp_netif_t *eth_netif = esp_netif_new(&eth_as_dhcps_cfg);
+    // Attach Ethernet driver to TCP/IP stack
+    ESP_ERROR_CHECK(esp_netif_attach(eth_netif, esp_eth_new_netif_glue(eth_handles[0])));
+    // Register user defined event handers
+    esp_event_handler_register(ETH_EVENT, ETHERNET_EVENT_CONNECTED, my_event_connected_handler, eth_netif);
+    esp_event_handler_register(IP_EVENT, IP_EVENT_AP_STAIPASSIGNED, my_ip_assigned_handler, NULL);
+    esp_event_handler_register(IP_EVENT, IP_EVENT_ETH_GOT_IP, &got_ip_event_handler, NULL);
```

Comment:
Nicpick, just to look the same as other handlers.
```suggestion
    esp_event_handler_register(IP_EVENT, IP_EVENT_ETH_GOT_IP, got_ip_event_handler, NULL);
```